app/controllers/buyers/users_controller.rb

```diff
@@ -2,29 +2,45 @@ class Buyers::UsersController < Buyers::BaseController
   activate_menu :audience, :accounts, :listing
 
   inherit_resources
-  actions :index, :show, :edit, :update, :destroy
+  actions :index, :show, :edit
   defaults :route_prefix => 'admin_buyers' #FIXME: inherited_resource makes us repeat this
   belongs_to :account, :collection_name => :buyer_accounts
 
   def update
     # TODO: I think this controller is used only on provider side
-    resource.validate_fields! if current_user.account.buyer?
-    resource.attributes = params[:user]
-    super
+    resource.validate_fields! if current_account.buyer?
+    resource.attributes = params[:user] # TODO: we should filter the params
```

Comment:
Why do we need to set the attributes here? Is this because of extra fields somehow? It does not look safe anyway.

Reply:
> Why do we need to set the attributes here?

Absolutely no idea 🤷‍♀️ but in case of doubt, I prefer not touching and keeping exactly the same behavior.

> t does not look safe anyway.

It's not safe but it is not changing anything of that from before this PR and not introduced just now.
Looking at which params must be filtered can be a task on its own.